src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import { generateId } from "../../utils";
+
+interface Props {
+  errorMessage?: string;
+  fieldName: string;
+  initialValue?: string;
+  inputError?: boolean;
+  inputFilter?: RegExp;
+  inputSuccess?: boolean;
+  label: string;
+  placeholder?: string;
+  prefix?: string;
+  required?: boolean;
+  suffix?: string;
+}
+/**
+ * TextInput Component
+ * @param {string}  label          Field label.
+ * @param {string}  fieldName      Name of the input field.
+ * @param {string}  [errorMessage] Error message text displayed when inputError === true.
+ * @param {string}  [initialValue] Optional default input value.
+ * @param {boolean} [inputError]   Triggers error message and error styling.
+ * @param {boolean} [inputSuccess] Trigger success styling.
+ * @param {string}  [placeholder]  Input field placeholder text.
+ * @param {string}  [prefix]       Text to be displayed at the front of input field. Not stored in value. Ex: currency indicator.
+ * @param {boolean} [required]     Adds semantic required attr and appends an * to the end of the input label.
+ * @param {string}  [suffix]       Text to be displayed at the end of input field. Not stored in value. Ex: mass indicator (lbs, fl oz)
```

Comment:
This is a pretty good idea to document in the component. I was thinking you could include these param documentations in the textInput.stories.tsx argTypes, which storybook will take an populate this section
`  argTypes: {
    label: {
      defaultValue: null,
      description: "Field label",
    },
  },`

![Screen Shot 2022-07-25 at 2 05 20 PM](https://user-images.githubusercontent.com/55826377/180873935-c48843d2-c7b6-4334-a8b1-97785d8b6562.png)

Reply:
Definitely can do!